src/Poseidon/CLI/Genoconvert.hs

```diff
@@ -42,89 +44,119 @@ data GenoconvertOptions = GenoconvertOptions
-                        _  -> liftIO . throwIO $ PoseidonGenericException ("Illegal outFormat " ++ outFormat ++ ". Only Outformats EIGENSTRAT or PLINK are allowed at the moment")
-                let genotypeData = GenotypeDataSpec gFileSpec (genotypeSnpSet . posPacGenotypeData $ pac)
-                    newPac = pac { posPacGenotypeData = genotypeData }
-                logInfo "Adjusting POSEIDON.yml..."
-                liftIO $ writePoseidonPackage newPac
-            -- delete now replaced input genotype data
-            let filesToDelete = case genotypeFileSpec . posPacGenotypeData $ pac of
-                    GenotypeEigenstrat g _ s _ i _ -> [g, s, i]
-                    GenotypePlink      g _ s _ i _ -> [g, s, i]
-                    GenotypeVCF        g _         -> [g]
-            when removeOld . liftIO . mapM_ (removeFile . (posPacBaseDir pac </>)) $ filesToDelete
+        logInfo "Processing SNPs..."
+        logA <- envLogAction
+        currentTime <- liftIO getCurrentTime
+        errLength <- envErrorLength
+        let eigenstratIndEntries = jannoRows2EigenstratIndEntries . posPacJanno $ pac
+        let zipEnding = if outZip then ".gz" else "" -- we need this to trigger the zipping
+        liftIO $ catch (
+            runSafeT $ do
+                eigenstratProd <- loadGenotypeData (posPacBaseDir pac) (posPacGenotypeData pac)
+                let outConsumer = case outFormat of
+                        "EIGENSTRAT" -> writeEigenstrat (outG ++ ".gconvert" ++ zipEnding)
+                                                        (outS ++ ".gconvert" ++ zipEnding)
+                                                        (outI ++ ".gconvert")
+                                                        eigenstratIndEntries
+                        "PLINK"      -> writePlink (outG ++ ".gconvert" ++ zipEnding)
+                                                   (outS ++ ".gconvert" ++ zipEnding)
+                                                   (outI ++ ".gconvert")
+                                                   (map (eigenstratInd2PlinkFam outPlinkPopMode) eigenstratIndEntries)
+                        _  -> liftIO . throwIO . PoseidonGenericException $
+                            "Illegal outFormat " ++ outFormat ++
+                            ". Only Outformats EIGENSTRAT or PLINK are allowed at the moment"
+                runEffect $ eigenstratProd >-> printSNPCopyProgress logA currentTime >-> outConsumer
+            ) (throwIO . PoseidonGenotypeExceptionForward errLength)
+        -- the following will just override if the file already exists.
+        liftIO $ renameFile (outG ++ ".gconvert" ++ zipEnding) outG
+        liftIO $ renameFile (outS ++ ".gconvert" ++ zipEnding) outS
+        liftIO $ renameFile (outI ++ ".gconvert") outI
+        logInfo "Done"
+        -- overwrite genotype data field in POSEIDON.yml file
+        unless (onlyGeno || isJust outPath) $ do
```

Comment:
We discussed at some point that these options are redundant. And their command line documentation is not very helpful. Maybe we should remove `--onlyGeno` and describe the effect of `-o` a bit better.

Reply:
Ja, können wir machen. Wäre dann halt streng-genommen ein breaking change

Reply:
Ok - you're right. I bumped to 1.6.0.0 now instead of 1.5.8.0. Version numbers are for free, after all.